ngraph/core/reference/include/ngraph/runtime/reference/einsum.hpp

```diff
@@ -0,0 +1,1210 @@
+                                                       matmul_operand2->get_data_ptr<T>(),
+                                                       matmul_output->get_data_ptr<T>(),
+                                                       matmul_operand1->get_shape(),
+                                                       matmul_operand2->get_shape(),
+                                                       matmul_output_shape,
+                                                       transpose_a,
+                                                       transpose_b);
+
+                    // step 4. reshape back by unrolling dimensions corresponding to separate labels
+                    // if needed now dimensions corresponding to reduced labels are reduced by the
+                    // MatMul operation
+                    common_part = "";
+                    for (const auto& common_label : common_labels)
+                    {
+                        common_part += common_label;
+                    }
+                    std::string resultant_subscript = common_part + separate_part1 + separate_part2;
+                    Shape back_shape;
+                    back_shape.insert(
+                        back_shape.end(), common_sub_shape.begin(), common_sub_shape.end());
+                    back_shape.insert(
+                        back_shape.end(), separate1_sub_shape.begin(), separate1_sub_shape.end());
+                    back_shape.insert(
+                        back_shape.end(), separate2_sub_shape.begin(), separate2_sub_shape.end());
+
+                    HostTensorPtr contract_output = std::shared_ptr<HostTensor>(
+                        new HostTensor(matmul_output->get_element_type(), back_shape));
+                    const AxisVector order = get_default_order(matmul_output->get_shape());
+                    ngraph::runtime::reference::reshape(
+                        reinterpret_cast<const char*>(matmul_output->get_data_ptr<T>()),
+                        reinterpret_cast<char*>(contract_output->get_data_ptr<T>()),
+                        matmul_output->get_shape(),
+                        order,
+                        back_shape,
+                        matmul_output->get_element_type().size());
+
+                    update_operands(inputs,
+                                    input_subscripts,
+                                    input_ind1,
+                                    input_ind2,
+                                    contract_output,
+                                    resultant_subscript);
+                }
+
+                template <typename T>
+                void einsum_impl(const HostTensorVector& inputs,
+                                 const HostTensorVector& outputs,
+                                 const std::string& equation)
+                {
+                    // parse equation to extract input subscripts and output subscript
```

Comment:
is this comment really needed? I propose to remove this line. Code a line above in my opinion is quite self-explanatory

Reply:
Fixed.